### PR TITLE
tkdiff: init at 5.6

### PR DESCRIPTION
**Pull request by ghost**
## Description of changes

TKdiff is a graphical diff that's been around for 20+ years. It is a script which depends on the tcl Tk toolkit and can integrate with tools like git, hg, etc.

https://tkdiff.sourceforge.io/

verified `git difftool` ran tkdiff and new files were properly shown against `/dev/null`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by robert-manchester** [comment]
Result of `nixpkgs-review pr 259340` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tkdiff</li>
  </ul>
</details>

**Comment by robert-manchester** [comment]
Result of `nixpkgs-review pr 259340` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tkdiff</li>
  </ul>
</details>

**Comment by robert-manchester** [comment]
Result of `nixpkgs-review pr 259340` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tkdiff</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -14912,6 +14912,12 @@
     githubId = 496447;
     name = "Robert Hensing";
   };
+  robert-manchester = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:14915]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/by-name/tk/tkdiff/package.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  patches = [ ./tkdiff.patch ];
+
+  nativeBuildInputs = [ unzip ];
+
+  buildInputs = [ tk ];
+
+  dontConfigure = true;
+  dontBuild = true;
+
```
Comment (SuperSandro2000) [inline pkgs/by-name/tk/tkdiff/package.nix:20]: If this doesn't break things, we don't need to disable it

File: pkgs/by-name/tk/tkdiff/package.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+stdenv.mkDerivation rec {
+  pname = "tkdiff";
+  version = "5.6";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/${pname}/${pname}-${builtins.replaceStrings ["."] ["-"] version}.zip";
+    hash = "sha256-Yox1QdSGmWtvmE9PcCzsfLOn/ZWdPocQQZK8PpRmILw=";
+  };
+
+  patches = [ ./tkdiff.patch ];
```
Comment (SuperSandro2000) [inline pkgs/by-name/tk/tkdiff/package.nix:12]: Why do we need this patch? A reason is missing

File: pkgs/by-name/tk/tkdiff/package.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
+{ lib, stdenv, fetchurl, tk, unzip }:
+
+stdenv.mkDerivation rec {
+  pname = "tkdiff";
+  version = "5.6";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/${pname}/${pname}-${builtins.replaceStrings ["."] ["-"] version}.zip";
```
Comment (SuperSandro2000) [inline pkgs/by-name/tk/tkdiff/package.nix:8]:
```suggestion
    url = "mirror://sourceforge/tkdiff/tkdiff-${builtins.replaceStrings ["."] ["-"] version}.zip";
```
it is nice to not replace pname wherever possible

File: pkgs/by-name/tk/tkdiff/package.nix [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+  buildInputs = [ tk ];
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm755 -t $out/bin ${pname}
```
Comment (SuperSandro2000) [inline pkgs/by-name/tk/tkdiff/package.nix:24]:
```suggestion
    install -Dm755 -t $out/bin tkdiff
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when naming your commits.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (robert-manchester) [reply maintainers/maintainer-list.nix:14915]: that is the order returned by sort, somewhat surprisingly, using various options (-d, -g, -h, no option).

**Comment by robert-manchester** [comment]
> Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when naming your commits.

thanks for the review! all comments addressed except for the order in maintainers as that is the order returned by sort.
```
sort <<EOF
roberth
robert-manchester
robertodr
EOF
```
```
roberth
robert-manchester
robertodr
```

**Comment by robert-manchester** [comment]
Result of `nixpkgs-review pr 259340` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tkdiff</li>
  </ul>
</details>

File: pkgs/by-name/tk/tkdiff/package.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
+{ diffutils, fetchurl, lib, makeBinaryWrapper, stdenv, tk, unzip }:
+
+stdenv.mkDerivation {
+  pname = "tkdiff";
+  version = "5.6-r189";
+
+  src = fetchurl {
+    # 5.6 + fix to make "tkdiff /dev/null file" work again.
```
Comment (SuperSandro2000) [inline pkgs/by-name/tk/tkdiff/package.nix:8]: Only from the version number I would have guessed that this is a stable release. Can we maybe download the stable release and apply the fix as a patch?

Comment (robert-manchester) [reply pkgs/by-name/tk/tkdiff/package.nix]: thanks! I changed version -> `5.6-patch-189`, do you have a better suggestion? added the patch to this change to patch the stable version.

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/by-name/tk/tkdiff/package.nix (hunk 6 above)
Comment (SuperSandro2000) [reply pkgs/by-name/tk/tkdiff/package.nix]: should be good enough

Comment (fgaz) [reply pkgs/by-name/tk/tkdiff/package.nix]: I think we can avoid the suffix. The patch is just  a bugfix, and we don't usually modify the version because of those.

Comment (robert-manchester) [reply pkgs/by-name/tk/tkdiff/package.nix]: updated and squashed commits.

File: pkgs/by-name/tk/tkdiff/package.nix [hunk 7]
```diff
@@ -0,0 +1,43 @@
+{ diffutils, fetchurl, lib, makeBinaryWrapper, stdenv, tk, unzip }:
+
+stdenv.mkDerivation rec {
+  pname = "tkdiff";
+  version = "5.6";
+
+  src = fetchurl {
```
[the remark this answers is not in the document]
Comment (robert-manchester) [reply pkgs/by-name/tk/tkdiff/package.nix]: fixed!

File: pkgs/by-name/tk/tkdiff/package.nix [hunk 8]
```diff
@@ -0,0 +1,43 @@
+{ diffutils, fetchurl, lib, makeBinaryWrapper, stdenv, tk, unzip }:
+
+stdenv.mkDerivation rec {
```
[the remark this answers is not in the document]
Comment (robert-manchester) [reply pkgs/by-name/tk/tkdiff/package.nix]: fixed!

**Review by fgaz — APPROVED** [review]
looks good
